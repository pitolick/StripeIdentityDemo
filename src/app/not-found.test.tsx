import React from 'react'
import { render } from '@testing-library/react'
import DefaultLayout, { NotFoundLayout } from './not-found'

// Header・Footerをモック化
jest.mock('@/components/parts', () => ({
  Header: () => <></>,
  Footer: () => <></>,
}))

// HeadingLineをモック化
jest.mock('@/components/elements', () => ({
  HeadingLine: () => <></>,
}))

describe('DefaultLayoutコンポーネント', () => {
  it('正しくレンダリングされること', () => {
    const { container } = render(<DefaultLayout />)
    expect(container).toMatchSnapshot()
  })
})

describe('NotFoundLayoutコンポーネント', () => {
  it('正しくレンダリングされること', () => {
    const { container } = render(<NotFoundLayout />)
    expect(container).toMatchSnapshot()
  })
})
