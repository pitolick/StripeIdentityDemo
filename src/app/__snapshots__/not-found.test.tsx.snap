// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultLayoutコンポーネント 正しくレンダリングされること 1`] = `
<div>
  <main
    id="main"
  >
    <div
      class="container py-14"
    >
      <div>
        <p>
          リクエストされたページは見つかりませんでした。
          <br />
          お探しのページは削除されたか、
          <br />
          URLが変更された可能性があります。
        </p>
      </div>
    </div>
  </main>
</div>
`;

exports[`NotFoundLayoutコンポーネント 正しくレンダリングされること 1`] = `
<div>
  <main
    id="main"
  >
    <div
      class="container py-14"
    >
      <div>
        <p>
          リクエストされたページは見つかりませんでした。
          <br />
          お探しのページは削除されたか、
          <br />
          URLが変更された可能性があります。
        </p>
      </div>
    </div>
  </main>
</div>
`;
